src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -644,7 +653,8 @@ public void upload(
           meta.getUserMetadata(),
           uploadStreamInfo.left,
           presignedUrl,
-          session.getHttpClientKey());
+          session.getHttpClientKey(),
+          encMat != null && encMat.getQueryId() != null ? encMat.getQueryId() : queryId);
```

Comment:
you have a static method queryIdFromEncMatOr for this expression

Reply:
it's private in a separate package and class - I duplicated the method and used it in this class
I don't want to add unnecessary public method for that

but thanks to that I found also some other places not covered by passing queryId...